src/operator/controllers/intents_controller.go

```diff
@@ -102,6 +114,46 @@ func (r *IntentsReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
+func (r *IntentsReconciler) mapProtectedServicesToClientIntents(obj client.Object) []reconcile.Request {
+	protectedServices := obj.(*otterizev1alpha2.ProtectedServices)
+	logrus.Infof("Enqueueing client intents for protected services %s", protectedServices.Name)
+
+	intentsToReconcile := r.getIntentsToProtectedServices(protectedServices)
+	return r.mapIntentsToRequests(intentsToReconcile)
+}
+
+func (r *IntentsReconciler) mapIntentsToRequests(intentsToReconcile []otterizev1alpha2.ClientIntents) []reconcile.Request {
+	requests := make([]reconcile.Request, 0)
+	for _, clientIntents := range intentsToReconcile {
+		request := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      clientIntents.Name,
+				Namespace: clientIntents.Namespace,
+			},
+		}
+		requests = append(requests, request)
+	}
+	return requests
+}
+
+func (r *IntentsReconciler) getIntentsToProtectedServices(protectedServices *otterizev1alpha2.ProtectedServices) []otterizev1alpha2.ClientIntents {
```

Comment:
What happens if a server that once was a protected service will be removed from the list? Do we catch this scenario?

Reply:
Well, I found out we don't delete network policies when client intents are deleted, and this problem should be solved together. We should be able to list all network policies of servers in the namespace and delete the redundants

Reply:
We discussed that offline and this will be solved in another PR